code42_on_poll_connector.py

```diff
@@ -113,8 +123,10 @@ def handle_on_poll(self, param, action_result):
             self._save_artifacts_from_file_events(container_id, details, file_events)
 
         # Save last time of last alert for future polling
-        if not self._connector.is_poll_now() and details.get("createdAt"):
-            self._state["last_time"] = details["createdAt"]
+        last_created_at = details.get("createdAt")
```

Comment:
If you poll once and set checkpoint to, e.g., `2021-06-01` and then you use poll now for a specific set of alerts created on `2021-04-15`, then the checkpoint would be set back to April 15 and next time we auto-poll we'll create duplicate alerts (I think? or maybe best case we'll just fetch them unnecessarily)

Should we only update the checkpoint when `source_id is None`?

Reply:
`start_date` and `end_date` only get used if there is no checkpoint.

Reply:
interesting point about source id though... if you specify a source id and your checkpoint is past that point, you would not get your alerts...

Perhaps they should be exclusive, like you say